test/jdk/com/sun/net/httpserver/simpleserver/jwebserver/MaxRequestTimeTest.java

```diff
@@ -0,0 +1,206 @@
+ */
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLHandshakeException;
+import jdk.test.lib.Platform;
+import jdk.test.lib.net.SimpleSSLContext;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.util.FileUtils;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+import static java.lang.System.out;
+import static java.net.http.HttpClient.Builder.NO_PROXY;
+import static org.testng.Assert.*;
+
+/**
+ * This test confirms that the jwebserver does not wait indefinitely for
+ * a request to arrive.
+ *
+ * The jwebserver has a maximum request time of 5 seconds, which is set with the
+ * "sun.net.httpserver.maxReqTime" system property. If this threshold is
+ * reached, for example in the case of an HTTPS request where the server keeps
+ * waiting for a plaintext request, the server closes the connection. Subsequent
+ * requests are expected to be handled as normal.
+ *
+ * The test checks in the following order that:
+ *    1. an HTTP request is handled successfully,
+ *    2. an HTTPS request fails due to the server closing the connection
+ *    3. another HTTP request is handled successfully.
+ */
+public class MaxRequestTimeTest {
+    static final Path JAVA_HOME = Path.of(System.getProperty("java.home"));
+    static final String JWEBSERVER = getJwebserver(JAVA_HOME);
+    static final Path CWD = Path.of(".").toAbsolutePath().normalize();
+    static final Path TEST_DIR = CWD.resolve("MaxRequestTimeTest");
+    static final String LOOPBACK_ADDR = InetAddress.getLoopbackAddress().getHostAddress();
+    static final AtomicInteger PORT = new AtomicInteger();
+
+    static SSLContext sslContext;
+
+    @BeforeTest
+    public void setup() throws IOException {
+        if (Files.exists(TEST_DIR)) {
+            FileUtils.deleteFileTreeWithRetry(TEST_DIR);
+        }
+        Files.createDirectories(TEST_DIR);
+
+        sslContext = new SimpleSSLContext().get();
+        if (sslContext == null)
+            throw new AssertionError("Unexpected null sslContext");
+    }
+
+    @Test
+    public void testMaxRequestTime() throws Throwable {
+        final var sb = new StringBuffer();  // stdout & stderr
+        final var p = startProcess("jwebserver", sb);
+
+        sendHTTPSRequest();  // server expected to terminate connection
+        sendHTTPRequest();   // server expected to respond successfully
+        sendHTTPSRequest();  // server expected to terminate connection
+        sendHTTPRequest();   // server expected to respond successfully
+
+        p.destroy();
```

Comment:
Since the `sendXXXRequest()` methods above could potentially throw exceptions, do you think it would be better to destroy the process in a finally block? Or maybe manage the process start and destroy as testng before/after lifecycle methods?

Reply:
Good point, I'll update that. Thanks!